lisp/tests/LispAstSpec.hs

```diff
@@ -0,0 +1,361 @@
+      let env =
+            extendEnv "y" (VBool True) $
+              extendEnv "x" (VInt 5) emptyEnv
+      length env `shouldBe` 2
+
+    it "should shadow previous bindings with same name" $ do
+      let env =
+            extendEnv "x" (VInt 10) $
+              extendEnv "x" (VInt 5) emptyEnv
+      lookupEnv "x" env `shouldBe` Just (VInt 10)
+
+  describe "lookupEnv" $ do
+    it "should return Nothing for empty environment" $ do
+      lookupEnv "x" emptyEnv `shouldBe` Nothing
+
+    it "should find existing binding" $ do
+      let env = extendEnv "x" (VInt 42) emptyEnv
+      lookupEnv "x" env `shouldBe` Just (VInt 42)
+
+    it "should return Nothing for non-existing binding" $ do
+      let env = extendEnv "x" (VInt 42) emptyEnv
+      lookupEnv "y" env `shouldBe` Nothing
+
+    it "should find the most recent binding when shadowed" $ do
+      let env =
+            extendEnv "x" (VInt 20) $
+              extendEnv "x" (VInt 10) emptyEnv
+      lookupEnv "x" env `shouldBe` Just (VInt 20)
+
+    it "should find bindings in a larger environment" $ do
+      let env =
+            extendEnv "c" (VBool False) $
+              extendEnv "b" (VBool True) $
+                extendEnv "a" (VInt 1) emptyEnv
+      lookupEnv "a" env `shouldBe` Just (VInt 1)
+      lookupEnv "b" env `shouldBe` Just (VBool True)
+      lookupEnv "c" env `shouldBe` Just (VBool False)
+
+-- | Test suite for Evaluator monad
+evaluatorSpec :: Spec
+evaluatorSpec = do
+  describe "runEvaluator" $ do
+    it "should run a successful computation" $ do
+      let computation = return (VInt 42)
+      let (result, _) = runEvaluator computation emptyEnv
+      result `shouldBe` Right (VInt 42)
+
+    it "should preserve the initial environment" $ do
+      let computation = return VUnit
+      let initialEnv = extendEnv "x" (VInt 5) emptyEnv
+      let (_, finalEnv) = runEvaluator computation initialEnv
+      finalEnv `shouldBe` initialEnv
+
+  describe "throwEvalError" $ do
+    it "should create an error result" $ do
+      let computation = throwEvalError "test error" :: Evaluator Value
+      let (result, _) = runEvaluator computation emptyEnv
+      result `shouldBe` Left (ErrorMsg "test error")
+
+-- | Test suite for error handling helpers
+errorSpec :: Spec
+errorSpec = do
+  describe "mkError" $ do
+    it "should create an ErrorMsg" $ do
+      mkError "test" `shouldBe` ErrorMsg "test"
+      mkError "" `shouldBe` ErrorMsg ""
+
+  describe "liftError" $ do
+    it "should lift Right values unchanged" $ do
+      liftError "context" (Right (42 :: Int)) `shouldBe` Right 42
+
+    it "should add context to Left values" $ do
+      (liftError "parsing" (Left "syntax error") :: Either ErrorMsg Int) `shouldBe` Left (ErrorMsg "parsing: syntax error")
+
+-- | Test suite for built-in operators
+builtinSpec :: Spec
+builtinSpec = do
+  describe "arithmetic operators" $ do
+    it "should define builtinPlus" $ do
+      builtinPlus `shouldBe` VarName "+"
```

Comment:
`builtinPlus = VarName "+"` I feel like theses tests are also not doing much for us